primitives/tests/api.rs

```diff
@@ -0,0 +1,284 @@
+        TooPreciseError, UnknownDenominationError,
+    };
+}
+
+#[test]
+fn api_can_use_modules_from_crate_root() {
+    use bitcoin_primitives::{
+        block, locktime, merkle_tree, pow, script, sequence, transaction, witness,
+    };
+}
+
+#[test]
+fn api_can_use_types_from_crate_root() {
+    use bitcoin_primitives::{
+        Block, BlockHash, BlockHeader, BlockVersion, CompactTarget, OutPoint, Script, ScriptBuf,
+        Sequence, Transaction, TransactionVersion, TxIn, TxMerkleNode, TxOut, Txid, Witness,
+        WitnessCommitment, WitnessMerkleNode, Wtxid,
+    };
+}
+
+#[test]
+fn api_can_use_all_types_from_module_locktime() {
+    use bitcoin_primitives::locktime::relative::{
+        DisabledLockTimeError, IncompatibleHeightError, IncompatibleTimeError, LockTime,
+    };
+    use bitcoin_primitives::locktime::{absolute, relative};
+}
+
+#[test]
+fn api_can_use_all_types_from_module_script() {
+    use bitcoin_primitives::script::{
+        RedeemScriptSizeError, Script, ScriptBuf, ScriptHash, WScriptHash, WitnessScriptSizeError,
+    };
+}
+
+// `Debug` representation is never empty (C-DEBUG-NONEMPTY).
+#[test]
+fn api_all_non_error_types_have_non_empty_debug() {
+    macro_rules! check_debug {
+        ($($t:expr);* $(;)?) => {
+            $(
+                let debug = format!("{:?}", $t);
+                assert!(!debug.is_empty());
+            )*
+        }
+    }
+
+    // All the enums.
+    check_debug! {
+        absolute::LockTime::ZERO;
+        relative::LockTime::ZERO
+    };
+
+    // We abuse `Arbitrary` here to get a quick and dirty instance.
+    let ab: [u8; 32] = [0xab; 32];
+    let mut u = arbitrary::Unstructured::new(&ab);
+    let transaction = Transaction::arbitrary(&mut u).unwrap();
+
+    // All the structs.
+    check_debug! {
+        block::Block::<Unchecked>::arbitrary(&mut u).unwrap().assume_checked(None);
+        block::Block::<Unchecked>::arbitrary(&mut u).unwrap();
+        block::Header::arbitrary(&mut u).unwrap();
+        block::Version::arbitrary(&mut u).unwrap();
+        block::BlockHash::from_byte_array(BYTES);
+        block::WitnessCommitment::from_byte_array(BYTES);
+        merkle_tree::TxMerkleNode::from_byte_array(BYTES);
+        merkle_tree::WitnessMerkleNode::from_byte_array(BYTES);
+        pow::CompactTarget::from_consensus(0x1d00_ffff);
+        SCRIPT.as_script();
+        ScriptHash::from_script(&SCRIPT).unwrap();
+        WScriptHash::from_script(&SCRIPT).unwrap();
+        SCRIPT.clone();
+        Sequence::arbitrary(&mut u).unwrap();
+        Transaction::arbitrary(&mut u).unwrap();
+        TxIn::arbitrary(&mut u).unwrap();
+        TxOut::arbitrary(&mut u).unwrap();
+        OutPoint::arbitrary(&mut u).unwrap();
+        transaction.compute_txid();
+        transaction.compute_wtxid();
+        transaction.version;
+        Witness::arbitrary(&mut u).unwrap();
+        // ad: witness::Iter<'a>,
```

Comment:
why the `ad` comment here?

Reply:
Oh this is a hangover from when there were more than 26 and I used `ad` as a variable name. Your comment made me remember that I forgot to fix all the local var names after removing a bunch of stuff yesterday (during rebase). Thanks